src/credentials/CHIPCertFromX509.cpp

```diff
@@ -670,6 +670,170 @@ static CHIP_ERROR ConvertCertificate(ASN1Reader & reader, TLVWriter & writer)
     return err;
 }
 
+static CHIP_ERROR ExtractPubkey(ASN1Reader & reader, Crypto::P256PublicKey & pubkey)
```

Comment:
The proposed implementation seem complicated. The alternative would be:
1. Convert X509 to CHIP cert
2. Load CHIP Cert into ChipCertificateSet
3. Get all Certificate components from ChipCertifictaeData